src/MySqlConnector/Core/ResultSet.cs

```diff
@@ -38,7 +38,7 @@ public async Task ReadResultSetHeaderAsync(IOBehavior ioBehavior)
 				var firstByte = payload.HeaderByte;
 				if (firstByte == OkPayload.Signature)
 				{
-					var ok = OkPayload.Create(payload.Span, Session.SupportsDeprecateEof, Session.SupportsSessionTrack);
+					var ok = OkPayload.Create(payload.Span, Session.Context);
```

Comment:
I like bundling these together, but it could be done using a new interface that `ServerSession` implements. That would avoid allocating a second object every time a `ServerSession` is created. Changed here: https://github.com/mysql-net/MySqlConnector/pull/1497/commits/8466783f5647063092fc54dc47a43e2a3e19e16a.